azure/spoke/variables.tf

```diff
@@ -85,32 +89,155 @@ variable "subnets" {
         actions = list(string)
       })
     ), {})
+    associate_default_route_table            = optional(bool, true)
+    associate_default_network_security_group = optional(bool, true)
   }))
   default = {}
+}
+
+#########################
+# Network Security Group
+#########################
+
+variable "create_default_network_security_group" {
+  description = "Create a Network Security Group to associate with all subnets."
+  type        = bool
+  default     = true
+}
+
+variable "network_security_group_name" {
+  description = "Name of the default Network Security Group"
+  type        = string
+  default     = "default-nsg"
+}
+
+variable "nsg_rules_inbound" {
+  description = "A list of objects describing a rule inbound."
+  type = list(object({
```

Comment:
Change to a map of objects where the key is the name of the rule

Reply:
Will be addressed in Issue: https://github.com/ans-coe/terraform-modules/issues/94